### ISSUE TITLE
Use Zod-mocks to define snapshot tests for the viewer app

### ISSUE THREAD

**Issue by jhgilbert** [boombox-LMS/b5x#39]
NOTE: Depends on [#32](https://github.com/boombox-LMS/b5x/issues/32).

Once we've defined detailed Zod schemas for fragment JSON data, we can use those to write snapshot tests for the React fragment components. We haven't seen a ton of stability issues in the viewer app so far, so I'm marking this as nice-to-have for the moment, but once Boombox is headed toward prod, this feels like the bare minimum for ensuring that a certain component type doesn't unexpectedly go haywire.
